math/mathmore/src/complex_quartic.h

```diff
@@ -150,21 +150,14 @@ gsl_poly_complex_solve_quartic (double a, double b, double c, double d,
 
         disc = R2 - Q3;
 
-//       more numerical problems with this calculation of disc
-//LM - use this since it fixes case reported in issue #6900 when disc is approx 0
-//      (needs to add additional test cases where disc is ~ 0)
-        double CR2 = 729 * rcub * rcub;
-        double CQ3 = 2916 * qcub * qcub * qcub;
-        disc = (CR2 - CQ3) / 2125764.0;
-
 
         if (0 == R && 0 == Q)
           {
             u[0] = -rc / 3;
             u[1] = -rc / 3;
             u[2] = -rc / 3;
           }
-        else if (disc == 0)
+        else if (fabs(disc) < 1e-16)
```

Comment:
Why not `std::abs`? That will also do the right thing for a given type, no?

Reply:
The entire file is using `fabs`, which *is* the right function get the absolute value of a `double`. In my opinion, it's a matter of consistency, it's best to always adapt to the surrounding code style for localized fixes...

Reply:
That is right. I didn't look at the rest of the file